src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -1796,6 +1793,14 @@ public void updateTriggerConfigURL() {
             String triggerInformationMessage = MessageFormat.format(
                     "IOException when fetching dynamic trigger url: {0}", ioe.getMessage());
             triggerInformationAction.setErrorMessage(triggerInformationMessage);
+        } finally {
+            // Now that the dynamic project list has been loaded, we can "count down"
+            // the latch so that the EventListener thread can begin to process events.
+            if (projectListIsReady.getCount() > 0) {
+                logger.debug("Trigger config URL updated: {}; latch is currently {}; decrementing it.", job.getName(),
+                        projectListIsReady.getCount());
+            }
+            projectListIsReady.countDown();
```

Comment:
It looks like this was inside the above if block before, why the move outside of it?

Reply:
It was moved outside since we need to unblock the worker in any case, otherwise a wrong URL would leave the worker being stuck in the latch indefinitely.